admin/src/main/resources/io/buoyant/admin/js/src/delegator.js

```diff
@@ -14,29 +14,80 @@ define([
     }
 
     function renderAll(resp) {
-      $('.result').html(renderNode(resp));
+      $('.result').html(renderNode(resp, true));
     }
 
-    function renderNode(obj, weight){
+    function treeType(obj){
       switch(obj.type) {
-        case "delegate": obj.isDelegate = true; obj.child = renderNode(obj.delegate); break;
+        case "delegate": return treeType(obj.delegate);
+        case "transformation": return treeType(obj.tree);
+        case "alt":
+          for (var i = 0; i < obj.alt.length; i++) {
+            var subtreeType = treeType(obj.alt[i]);
+            if (subtreeType != "neg") {
+              return subtreeType;
+            }
+          }
+          return "neg";
+        case "union":
+          if (obj.union.some(function(e,_i){return treeType(e.tree) == "fail";})) {
+            return "fail";
+          } else if (obj.union.some(function(e,_i){return treeType(e.tree) == "success";})) {
+            return "success";
+          } else {
+            return "neg";
+          }
+        case "neg": return "neg";
+        case "fail": return "fail";
+        case "leaf": return treeType(obj.bound);
+        case "exception": return "fail";
+      }
+      if (obj.addr.type == "neg") {
+        return "neg";
+      } else {
+        return "success";
+      }
+    }
+
+    function renderNode(obj, primary, weight){
+      var ttype = treeType(obj);
+      switch(obj.type) {
+        case "delegate": obj.isDelegate = true; obj.child = renderNode(obj.delegate, primary); break;
```

Comment:
tioli this was already here, but for the record I prefer `break` to be on its own line